lib/nerves_runtime/kernel/uevent.ex

```diff
@@ -56,7 +56,7 @@ defmodule Nerves.Runtime.Kernel.UEvent do
 
   @impl true
   def handle_info({ref, result}, %State{discover_ref: ref} = s) do
-    Logger.debug("UEvent initial device discovery completed: #{result}")
+    _ = Logger.debug("UEvent initial device discovery completed: #{result}")
```

Comment:
I get this one a lot in my projects. I actually use `"-Wunmatched_returns"` for dialyzer flags.

Reply:
Yeah, this one annoyed me a bit too.

Turning off `unmatched_returns` causes several legit issues in this project to be ignored. I actually have it off because I didn't want to fix the issues before making the next release, but there's some problematic stuff that it caught. (Not the Logger ones)

I think we need to have `unmatched_returns` enabled. If there's an easy way of ignoring all Logger.debug calls, then I'd be for that.